cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -2009,20 +2013,50 @@ func (p *BicepProvider) ensureParameters(
 		// If the parameter is tagged with {type: "generate"}, skip prompting.
 		// We generate it once, then save to config for next attempts.`.
 		azdMetadata, hasMetadata := param.AzdMetadata()
-		if hasMetadata && parameterType == ParameterTypeString && azdMetadata.Type != nil &&
-			*azdMetadata.Type == azure.AzdMetadataTypeGenerate {
-
-			// - generate once
-			genValue, err := autoGenerate(key, azdMetadata)
-			if err != nil {
-				return nil, err
-			}
-			configuredParameters[key] = azure.ArmParameterValue{
-				Value: genValue,
+		if hasMetadata && parameterType == ParameterTypeString && azdMetadata.Type != nil {
```

Comment:
I am ok with the concept of azd metadata in bicep as long as it only provides a better azd experience for the user.  At the point where azd metadata is **required** for a feature that is where I would like to draw the line.  Any bicep that a user writes in my opinion should be able to run through a standard ARM/Bicep deployment without azd.

In this case it looks ok since I believe it is just setting some param value based on current logged in principal but i couldn't say the same for the generation scenario.